pkg/crinit/aggregated/aggregated.go

```diff
@@ -0,0 +1,399 @@
+	return nil
+}
+
+// createServiceAccount handles the creation of the service account for the
+// the cluster registry to be used with RBAC.
+func createServiceAccount(cmdOut io.Writer, clientset client.Interface,
+	opts *aggregatedClusterRegistryOptions) (*v1.ServiceAccount, error) {
+
+	sa := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccount,
+			Namespace: opts.ClusterRegistryNamespace,
+			Labels:    util.ComponentLabel,
+		},
+	}
+
+	if opts.DryRun {
+		return sa, nil
+	}
+
+	return clientset.CoreV1().ServiceAccounts(opts.ClusterRegistryNamespace).Create(sa)
+}
+
+// createClusterRole creates the cluster role for the operations we will allow
+// on our resources
+func createClusterRole(cmdOut io.Writer, clientset client.Interface,
+	opts *aggregatedClusterRegistryOptions) (*rbacv1.ClusterRole, error) {
+
+	rule := rbacv1.PolicyRule{
+		Verbs:     clusterRoleVerbs,
+		APIGroups: clusterRoleAPIGroup,
+		Resources: clusterRoleResources,
+	}
+
+	cr := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clusterRole,
+			Labels: util.ComponentLabel,
+		},
+		Rules: []rbacv1.PolicyRule{rule},
+	}
+
+	if opts.DryRun {
+		return cr, nil
+	}
+
+	return clientset.RbacV1().ClusterRoles().Create(cr)
+}
+
+// createClusterRoleBinding creates the cluster role binding for the operations we will allow
+// on our resources
+func createClusterRoleBindings(cmdOut io.Writer, clientset client.Interface,
+	opts *aggregatedClusterRegistryOptions) (*rbacv1.ClusterRoleBinding,
+	*rbacv1.ClusterRoleBinding, error, error) {
+
+	// cluster role binding for clusterregistry.k8s.io:apiserver cluster role
+	crb1 := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clusterRoleBindingAPIServer,
+			Labels: util.ComponentLabel,
+		},
+		Subjects: []rbacv1.Subject{
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				APIGroup:  "",
+				Name:      serviceAccount,
+				Namespace: opts.ClusterRegistryNamespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     clusterRole,
+		},
+	}
+
+	// cluster role binding for system:auth-delegator cluster role
+	crb2 := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clusterRoleBindingAuthDelegator,
+			Labels: util.ComponentLabel,
+		},
+		Subjects: []rbacv1.Subject{
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				APIGroup:  "",
+				Name:      serviceAccount,
+				Namespace: opts.ClusterRegistryNamespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "system:auth-delegator",
+		},
+	}
+
+	if opts.DryRun {
+		return crb1, crb2, nil, nil
+	}
+
+	result1, err1 := clientset.RbacV1().ClusterRoleBindings().Create(crb1)
+	result2, err2 := clientset.RbacV1().ClusterRoleBindings().Create(crb2)
+	return result1, result2, err1, err2
```

Comment:
I think it's a more standard pattern to return if the first call fails. So, this method would have a return signature `(*rbacv1.ClusterRoleBinding, *rbacv1.ClusterRoleBinding, error)`